packages/flutter/lib/src/services/sensitive_content.dart

```diff
@@ -0,0 +1,155 @@
+  // of `CONTENT_SENSITIVITY_AUTO` on Android that has implemented based on autofill hints; see
+  // https://github.com/flutter/flutter/issues/160879.
+  autoSensitive,
+
+  /// The widget tree contains sensitive content.
+  ///
+  /// When this level is set via a `SensitiveContent` widget, the window
+  /// hosting the screen will be marked as sensitive during an active media
+  /// projection session.
+  ///
+  /// See https://developer.android.com/reference/android/view/View#CONTENT_SENSITIVITY_SENSITIVE.
+  sensitive,
+
+  /// The widget tree does not contain sensitive content.
+  ///
+  /// When this level is set via a `SensitiveContent` widget, the window
+  /// hosting the screen will only be marked as sensitive if other `SensitiveContent`
+  /// widgets in the Flutter app with the [sensitive] level are present in the widget tree.
+  ///
+  /// See https://developer.android.com/reference/android/view/View#CONTENT_SENSITIVITY_NOT_SENSITIVE.
+  notSensitive,
+
+  /// The sensitivity content level is unknown to Flutter.
+  ///
+  /// This mode may represent the current content sensitivity of the window if, for example, Android
+  /// adds a new mode that is not recognized by the `SensitiveContent` widget.
+  ///
+  /// This mode cannot be used to set the sensitivity level of a `SensitiveContent` widget.
+  _unknown,
+}
+
+/// Service for setting the content sensitivity of the native app window (Android `View`)
+/// that contains the app's widget tree.
+///
+/// This service is only currently supported on Android API 35+.
+class SensitiveContentService {
+  /// Creates service to set content sensitivity of an app window (Android `View`) via
+  /// communication over the sensitive content [MethodChannel].
+  SensitiveContentService() {
+    sensitiveContentChannel = SystemChannels.sensitiveContent;
+  }
+
+  /// The channel used to communicate with the shell side to get and set the
+  /// content sensitivity of an app window (Android `View`).
+  late MethodChannel sensitiveContentChannel;
+
+  /// Sets content sensitivity level of the app window (Android `View`) that contains the app's widget
+  /// tree to the level specified by [contentSensitivity] via a call to the native embedder.
+  Future<void> setContentSensitivity(ContentSensitivity contentSensitivity) async {
+    try {
+      await sensitiveContentChannel.invokeMethod<void>(
+        'SensitiveContent.setContentSensitivity',
+        contentSensitivity.name,
+      );
+    } catch (e) {
+      // Content sensitivity failed to be set.
+      throw FlutterError('Content sensitivity $contentSensitivity failed to be set: $e');
+    }
+  }
+
+  /// Gets content sensitivity level of the app window (Android `View`) that contains
+  /// the app's widget tree.
+  Future<ContentSensitivity> getContentSensitivity() async {
+    String? result;
+    try {
+      result = await sensitiveContentChannel.invokeMethod<String>(
```

Comment:
Why are we using string intead of index of enum to communicate the sensitivity level?

Reply:
right now the communication is the following.

andoird (int) -> converting to String -> channel ->  converting string to enum -> dart(enum)

dart(enum) -> converting to String -> channel -> converting string to int -> andoird (int)


we could just do the following

android (int) -> converting to enum int in dart -> channel -> recast int to enum -> dart(enum)

dart(enum) -> recast to int -> channel -> converting to android int -> andoird (int)

Reply:
Isn't this the same level of work?

Reply:
Also, I think the `String`s are more readable for the serialization/deserialization done on the platform side.

Reply:
I think the piece I was missing is that sending `String`s through the channel is slightly more costly, thanks.

Changed to use the index of enums instead of the name. I didn't make a matching enum on the Android side to do this (I don't think it adds much value since there aren't many modes), but if you feel strongly, let me know.